// See route_guide

syntax = "proto3";

package rootgahyd;

service RootGahyd {
  rpc GetFeature(Point) returns (Feature) {}
  rpc ListFeatures(Rectangle) returns (stream Feature) {}
  rpc RecordRoot(stream Point) returns (RootSummary) {}
  rpc RootChat(stream RootNote) returns (stream RootNote) {}
}

message Point {
  int32 latitude = 1;
  int32 longitude = 2;
}

message Rectangle {
    Point lo = 1;
    Point hi = 2;
}

message Feature {
    string name = 1;
    Point location = 2;
}

message RootNote {
    Point location = 1;
    string message = 2;
}

message RootSummary {
    int32 point_count = 1;
    int32 feature_count = 2;
    int32 distance = 3;
    int32 elapsed_time = 4;
}
